designs/mixins.md

```diff
@@ -518,27 +569,52 @@ structure A1 {
 
 @mixin
 structure A2 {
-    A: String
+    A: Integer
 }
 
-structure Invalid with
-    A1
-    A2 {}
+structure Invalid with [A1, A2] {}
+```
+
+Members that are mixed into shapes MAY be redefined if and only if each
+redefined member targets the same shape. Traits applied to redefined members
+supersede any traits inherited from mixins.
+
+```
+@mixin
+structure A1 {
+    @private
+    a: String
+}
+
+@mixin
+structure A2 {
+    @required
+    a: String
+}
+
+structure Valid with [A1, A2] {}
 ```
 
 
 ### Mixins in the IDL
 
-To support mixins, `structure_statement` and `union_statement` ABNF rules
+To support mixins, shape ABNF rules
 will be updated to contain an optional `mixins` production
-that comes after the shape name and before `{`. Each shape ID referenced in
+that comes after the shape name. Each shape ID referenced in
 the `mixins` production MUST target a shape of the same type as the
 shape being defined and MUST be marked with the `@mixin` trait.
 
 ```
-structure_statement = "structure" ws identifier ws [mixins ws] structure_members
-union_statement = "union" ws identifier ws [mixins ws] union_members
-mixins = "with" 1*(ws shape_id)
+simple_shape_statement = simple_type_name ws identifier [mixins]
+list_statement = "list" ws identifier [mixins ws] shape_members
```

Comment:
`shape_members` doesn't have a leading `ws` token in the current spec:

```
shape_members =
    empty_shape_members / populated_shape_members

empty_shape_members =
    "{" ws "}"

populated_shape_members =
    "{" ws shape_member_kvp
    *(comma shape_member_kvp ws) trailing_comma "}"
```

So this, and others, needs ws outside of `[mixins]` right?

```
list_statement = "list" ws identifier [mixins] ws shape_members
```